Sources/Bot/Services/MergeService.swift

```diff
@@ -402,27 +402,36 @@ extension MergeService {
                     .observe(on: scheduler)
             case .blocked,
                  .unstable:
-                return github.fetchCommitStatus(for: metadata.reference)
-                    .flatMap(.latest) { commitStatus -> SignalProducer<Event, AnyError> in
-                        switch commitStatus.state {
-                        case .pending:
-                            return .value(.integrationDidChangeStatus(.updating, metadata))
-                        case .failure:
-                            return .value(.integrationDidChangeStatus(.failed(.checksFailing), metadata))
-                        case  .success:
-                            return github.fetchPullRequest(number: metadata.reference.number)
-                                .map { metadata in
-                                    switch metadata.mergeState {
-                                    case .clean:
-                                        return .retryIntegration(metadata)
-                                    default:
-                                        return .integrationDidChangeStatus(.failed(.blocked), metadata)
+                let pullRequest = metadata.reference
+                return github.fetchAllStatusChecks(for: pullRequest).map { statusChecks -> Bool in
+                    return statusChecks.map({ $0.state }).contains(.pending)
```

Comment:
for some reason github reports checks not as I expect, on ci they are called "test" and "build" but when I tested it this request returned one completed check (one that was run) and two pending checks, all with name "build" 🤷‍♂ 
I mean it works for what we need to do here, but is unexpected

Reply:
I think every CI run is a `build`, there are 2 types of status checks on GH: checks and statuses. Status is the "old" checks, which may support less types. Checks is the tab we tried in the temp repo, it has more rich features (and probably more CI job types as well)

Reply:
Note that if the bot is configured to ignore failures from non-required GitHub checks, we might consider that we don't care if those non-required GitHub checks are still pending and could consider merging the PR as soon as all required checks are green even if non-required are still orange.

That can be for a separate ticket and PR though, but I think it's worth adding a `// TODO: [IOSP-xxx]` comment in there to not forget.

```suggestion
                    // TODO: [IOSP-???] Only wait for pending checks that are _required_ if bot is configured to ignore non-required checks
                    return statusChecks.map({ $0.state }).contains(.pending)
```